pumps/influx2.go

```diff
@@ -0,0 +1,236 @@
+package pumps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/influxdata/influxdb-client-go/v2/domain"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+)
+
+type Influx2Pump struct {
+	dbConf *Influx2Conf
+	client influxdb2.Client
+	CommonPumpConfig
+}
+
+var (
+	influx2Prefix      = "influx2-pump"
+	influx2DefaultENV  = PUMPS_ENV_PREFIX + "_INFLUX2" + PUMPS_ENV_META_PREFIX
+	influx2Measurement = "analytics"
+)
+
+type RetentionRule struct {
+	EverySeconds              int64  `mapstructure:"every_seconds" json:"every_seconds"`
+	ShardGroupDurationSeconds int64  `mapstructure:"shard_group_duration_seconds" json:"shard_group_duration_seconds"`
+	Type                      string `mapstructure:"type" json:"type"`
+}
+
+type NewBucket struct {
+	Description    string          `mapstructure:"description" json:"description"`
+	RetentionRules []RetentionRule `mapstructure:"retention_rules" json:"retention_rules"`
```

Comment:
Can you add some comments to this field?

Reply:
I've added some, but they're not quite as detailed as I'd like. The InfluxDB2 API documentation is a little bit lacking in this area unfortunately.